internal/middleware/perfinsights/doc.go

```diff
@@ -0,0 +1,7 @@
+// Package perfinsights defines middleware that reports the latency of API calls to Prometheus.
+//
+// Unlike the gRPC middleware, the perf insights middleware records API calls by "shape", versus
+// aggregating all calls simply by API kind. The shapes allow for more granular determination of
```

Comment:
Nitpick: to me, "shape" and "kind" are kind of synonyms. Might be good to be define "shape"